app/controllers/application_controller.rb

```diff
@@ -1,8 +1,8 @@
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::Base
-  rescue_from StandardError, with: :respond_with_error
-
+  rescue_from StandardError, ActiveRecord::RecordNotFound, with: :respond_with_error
+  rescue_from ActiveRecord::RecordNotFound, with: :respond_with_not_found
```

Comment:
Die zweite Zeile mit dem eigenen Respond für `ActiveRecord::RecordNotFound` wird in dieser Konstellation nie erreicht. Das erste Rescue fängt den Fehler und "behandelt" ihn.